Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -126,75 +199,43 @@ public States States
         /// </summary>
         public string VoiceCommand = "";
 
+        [FormerlySerializedAs("RequiresFocus")]
+        [SerializeField]
+        public bool voiceRequiresFocus = true;
         /// <summary>
         /// Does the voice command require this to have focus?
         /// Registers as a global listener for speech commands, ignores input events
         /// </summary>
-        public bool RequiresFocus = true;
-
-        /// <summary>
-        /// Does this interactable require focus
-        /// </summary>
-        public bool FocusEnabled { get { return !IsGlobal; } set { IsGlobal = !value; } }
-
-        /// <summary>
-        /// Event receivers can be used to listen for different
-        /// events at runtime. This method allows receivers to be dynamically added at runtime.
-        /// </summary>
-        /// <returns>The new event receiver</returns>
-        public T AddReceiver<T>() where T : ReceiverBase, new()
-        {
-            var interactableEvent = new InteractableEvent();
-            var result = new T();
-            result.Event = interactableEvent.Event;
-            interactableEvent.Receiver = result;
-            InteractableEvents.Add(interactableEvent);
-            return result;
-        }
-
-        /// <summary>
-        /// Returns the first receiver of type T on the interactable,
-        /// or null if nothing is found.
-        /// </summary>
-        public T GetReceiver<T>() where T : ReceiverBase
+        public bool VoiceRequiresFocus
         {
-            for (int i = 0; i < InteractableEvents.Count; i++)
+            get { return voiceRequiresFocus; }
+            set
             {
-                if (InteractableEvents[i] != null && InteractableEvents[i].Receiver is T)
+                if (voiceRequiresFocus != value)
                 {
-                    return (T) InteractableEvents[i].Receiver;
-                }
-            }
-            return null;
-        }
+                    voiceRequiresFocus = value;
 
-        /// <summary>
-        /// Returns all receivers of type T on the interactable.
-        /// If nothing is found, returns empty list.
-        /// </summary>
-        public List<T> GetReceivers<T>() where T : ReceiverBase
-        {
-            List<T> result = new List<T>();
-            for (int i = 0; i < InteractableEvents.Count; i++)
-            {
-                if (InteractableEvents[i] != null && InteractableEvents[i].Receiver is T)
-                {
-                    result.Add((T)InteractableEvents[i].Receiver);
+                    // If we are active, then change global speech registeration. 
+                    // Register handle if we do not require focus, unregister otherwise
+                    if (gameObject.activeInHierarchy)
+                    {
```

Comment:
why is this check required?

Reply:
Updated comment.

Basically, if we are disabled (GameObject level), then we will re-register ourselves during OnEnable because we generally unregister ourselves in the OnDisable

This is to ensure Interactable doesn't respond to input when disabled